File: LeetCode/cpp/75.cpp
class Solution {
public:
    void sortColors(vector<int> &nums) {
        int i = 0, j = 0, k = nums.size() - 1;
        while (j <= k) {
            if (nums[j] == 2) swap(nums[j], nums[k--]);
            else if (nums[j] == 0) swap(nums[i++], nums[j++]);
            else
                ++j;
        }
    }
};

File: LeetCode/python3/75.py
class Solution:
    def sortColors(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i, j, k = 0, 0, len(nums)-1
        while j <= k:
            if nums[j] == 0:
                nums[j], nums[i] = nums[i], nums[j]
                i += 1
                j += 1
            elif nums[j] == 2:
                nums[j], nums[k] = nums[k], nums[j]
                k -= 1
            else:
                j += 1
